public/pages/ComposableTemplates/containers/ComposableTemplates/ComposableTemplates.tsx

```diff
@@ -38,11 +38,15 @@ import ComponentTemplateBadge from "../../../../components/ComponentTemplateBadg
 import AssociatedTemplatesModal from "../AssociatedTemplatesModal";
 import { useComponentMapTemplate } from "../../utils/hooks";
 import "./index.scss";
+import { DataSourceMenuContext } from "../../../../services/DataSourceMenuContext";
 
 interface ComposableTemplatesProps extends RouteComponentProps {
   commonService: CommonService;
   componentMapTemplate: Record<string, string[]>;
   loading: boolean;
+  dataSourceId: string;
+  dataSourceLabel: string;
+  multiDataSourceEnabled: boolean;
```

Comment:
this props is used for?

Reply:
dataSourceId & dataSourceLabel are used for setting the state and updating the URL. While `multiDataSourceEnabled` props is not currently used in this component, I see a potential need for it as to hide dataSourceId & dataSourceLabel as query params from the URL.